### PR TITLE
Add the ability to specify additional exposed ports for @QuarkusIntegrationTest

### PR DESCRIPTION
**Pull request by geoand**
Originally mentioned in https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/QuarkusIntegrationTest.2C.20container.20config

Resolves; #23722 23722

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM.

I haven't fully tested, but should any check be performed if the additional exposed port is being used or will it fail when that happens?

**Comment by geoand** [comment]
It won't fail, but the log which displays the launch command should make it easy to understand what's going on, if anyone gets into trouble.

File: test-framework/common/src/main/java/io/quarkus/test/common/DefaultDockerContainerLauncher.java [hunk 1]
```diff
@@ -97,6 +99,10 @@ public void start() throws IOException {
         args.add(httpPort + ":" + httpPort);
         args.add("-p");
         args.add(httpsPort + ":" + httpsPort);
+        for (var entry : additionalExposedPorts.entrySet()) {
+            args.add("-p");
+            args.add(entry.getKey() + ":" + entry.getValue());
```
Comment (sdaclin) [inline test-framework/common/src/main/java/io/quarkus/test/common/DefaultDockerContainerLauncher.java:104]: Shouldn't the external port be dynamically assigned like for http port ?

Comment (geoand) [reply test-framework/common/src/main/java/io/quarkus/test/common/DefaultDockerContainerLauncher.java:104]: The configuration allows you to set the host and mapped port. Why would you want to dynamically assign the host port?

**Pull request merged**